### PR TITLE
dev: Add static-analysis page

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: dev/source/docs/static-analysis.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
+.. _static-analysis:
+
+===============
+Static Analysis
+===============
+
+Two tools are often used by ArduPilot to perform static analysis on the codebase, clang's "scan build" option and Coverity.
```
Comment (Hwurzburg) [inline dev/source/docs/static-analysis.rst:7]: Coverity does not seem to be shown

File: dev/source/docs/static-analysis.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+   You may need to install clang for this to work
+
+Afterwards you can view the results (e.g.):
+
+::
+
+   scan-view /home/pbarker/rc/ardupilot/tmp/scan-build-2020-08-11-114140-28761-1
+
+On Ubuntu 18.04 I actually need:
```
Comment (Hwurzburg) [inline dev/source/docs/static-analysis.rst:29]: third person please

File: dev/source/docs/static-analysis.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+You can do your own static analysis using autotest:
+
+::
+
+   ./Tools/autotest/autotest.py clang-scan-build
+
+.. note::
+
+   You may need to install clang for this to work
```
Comment (Hwurzburg) [inline dev/source/docs/static-analysis.rst:21]: how does one install clang?

File: dev/source/docs/static-analysis.rst [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+Afterwards you can view the results (e.g.):
+
+::
+
+   scan-view /home/pbarker/rc/ardupilot/tmp/scan-build-2020-08-11-114140-28761-1
```
Comment (Hwurzburg) [inline dev/source/docs/static-analysis.rst:27]: somehow make path generic?

**Comment by peterbarker** [comment]
On Tue, 11 Aug 2020, Henry Wurzburg wrote:

> +Two tools are often used by ArduPilot to perform static analysis on the codebase, clang's "scan build" option and Coverity.
> 
> Coverity does not seem to be shown

Yes, that's because I can't remember how we interact with that and how 
useful it is to others.  That's room for future expansion...

> +On Ubuntu 18.04 I actually need:
> 
> third person please

First person was quite deliberate - I don't know how general a problem it 
is.  We could cut this section, but it may help someone else.

> +   You may need to install clang for this to work
> 
> how does one install clang?

That's platform and distribution dependent.  Perhaps we have docs 
somewhere else on how that's done?

> +   scan-view /home/pbarker/rc/ardupilot/tmp/scan-build-2020-08-11-114140-28761-1
> 
> somehow make path generic?

That may be able to be done with scan-view 
$PWD/tmp/scan-build-2020-08-11-114140-28761-1 - but that's not going to be 
generic as the last bit it generated.  It does give you the command to run 
as part of the output.

File: dev/source/docs/static-analysis.rst (hunk 1 above)
Comment (khancyr) [reply dev/source/docs/static-analysis.rst:7]: Coverity is a bit tricky as it is done with a cron job on travis

File: dev/source/docs/static-analysis.rst (hunk 2 above)
Comment (Hwurzburg) [reply dev/source/docs/static-analysis.rst:29]: Suggested change: "On Ubuntu 18.04 one might also need:"

File: dev/source/docs/static-analysis.rst (hunk 3 above)
Comment (Hwurzburg) [reply dev/source/docs/static-analysis.rst:21]: Unbuntu: sudo apt-get install clang
Windows: https://clang.llvm.org/get_started.html

File: dev/source/docs/static-analysis.rst (hunk 4 above)
Comment (Hwurzburg) [reply dev/source/docs/static-analysis.rst:27]: Suggested:
scan-view /home/<user>/<path to repo>/ardupilot/https://clang.llvm.org/get_started.html

File: dev/source/docs/static-analysis.rst (hunk 3 above)
Comment (peterbarker) [reply dev/source/docs/static-analysis.rst]: Done

File: dev/source/docs/static-analysis.rst (hunk 4 above)
Comment (peterbarker) [reply dev/source/docs/static-analysis.rst]: Adjusted it.

File: dev/source/docs/static-analysis.rst (hunk 2 above)
Comment (peterbarker) [reply dev/source/docs/static-analysis.rst]: Changed.

**Pull request merged**